solidity/src/base/Constants.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+uint256 constant WORD_SIZE = 0x20;
+uint256 constant WORDX2_SIZE = 0x40;
+uint256 constant WORDX3_SIZE = 0x60;
+uint256 constant WORDX4_SIZE = 0x80;
+uint256 constant WORDX5_SIZE = 0xA0;
+uint256 constant WORDX6_SIZE = 0xC0;
+uint256 constant WORDX7_SIZE = 0xE0;
+uint256 constant WORDX8_SIZE = 0x100;
+uint256 constant WORDX9_SIZE = 0x120;
+uint256 constant WORDX10_SIZE = 0x140;
+uint256 constant WORDX11_SIZE = 0x160;
+uint256 constant WORDX12_SIZE = 0x180;
+
+uint256 constant ECADD_ADDRESS = 0x06;
+uint256 constant ECMUL_ADDRESS = 0x07;
+uint256 constant ECPAIRING_ADDRESS = 0x08;
+uint256 constant ECADD_GAS = 150;
+uint256 constant ECMUL_GAS = 6000;
+uint256 constant ECPAIRINGX2_GAS = 113000;
```

Comment:
isn't the gas for ecPairing = 45000 ?

also, why do we need 2x amount the gas?

Reply:
is it because we pass 12 words in line 70?

Reply:
It's 45000 + 34000 * number of pairings. So, this is the gas for 2 pairings.